frontend/src/components/NavBar.js

```diff
@@ -73,64 +74,103 @@ export default function NavBar() {
-        <input
-          type="checkbox"
-          id="replay"
-          checked={replay}
-          onChange={handleCheckBox}
-        />
-        <span className="ml-1 text-white">Replay</span>
-      </div>
-      {replay ? (
-        <>
-          <button
-            className="bg-blue-600 hover:bg-blue-800 text-white text-bold py px-4 rounded ml-8"
-            type="button"
-            onClick={moveBackward}
-          >
-            Prev
-          </button>
-          <button
-            className="bg-blue-600 hover:bg-blue-800 text-white text-bold py px-4 rounded ml-8"
-            type="button"
-            onClick={compile}
-          >
-            Run
-          </button>
-          <button
-            className="bg-blue-600 hover:bg-blue-800 text-white text-bold py px-4 rounded ml-8"
-            type="button"
-            onClick={moveForward}
-          >
-            Next
-          </button>{" "}
-        </>
-      ) : (
-        <button
-          className="bg-blue-600 hover:bg-blue-800 text-white text-bold py px-4 rounded ml-8"
-          type="button"
-          onClick={compile}
-        >
-          Play
-        </button>
-      )}
-    </nav>
+    <div className="w-full bg-gray-800 border-b border-gray-600">
+      <nav className="h-12 w-1/2 justify-between flex p-2 border-r border-gray-600">
+        {replay ? (
+          <div className="flex justify-between mt-1 w-40">
+            <StepButton left onClick={moveBackward}>
+              <Vertical left />
+              <Arrow left />
+            </StepButton>
+            <PlayButton onClick={compile} />
+            <StepButton type="button" onClick={moveForward}>
+              <Arrow />
+              <Vertical />
+            </StepButton>
+          </div>
+        ) : (
+          <PlayButton onClick={compile} />
+        )}
+        <div className="flex">
+          <div className="mr-2 pt-1">
+            <input
+              type="checkbox"
+              id="replay"
+              checked={replay}
+              onChange={handleCheckBox}
+            />
+            <span className="ml-1 text-white">Replay</span>
+          </div>
+          <div className="pb-5 ml-2">
+            <input
+              style={{ display: "none" }}
+              type="file"
+              ref={hiddenFileInput}
+              onChange={handleOnChange}
+            />
+            <button
+              className="bg-blue-600 hover:bg-blue-800 text-white py-1 px-2 rounded"
+              type="button"
+              onClick={handleOnClick}
+            >
+              Upload a File
+            </button>
+          </div>
+        </div>
+      </nav>
+    </div>
   );
 }
+
+const PlayButton = styled.button`
+  width: 0;
+  height: 24px;
+  border-style: solid;
+  border-color: transparent transparent transparent #d3d3d3;
+  box-sizing: border-box;
+  border-width: 12px 0px 12px 18px;
+  transition: 200ms all ease;
+  margin: 0 10px 0 10px;
```

Comment:
this can be shortened to `margin: 0 10px`

Reply:
Okie!